doc/api/ooni/orchestrate.md

```diff
@@ -0,0 +1,233 @@
+
+std::string production_events_url();
+std::string testing_events_url();
+
+class Client {
+  public:
+    Var<Logger> logger = Logger::global();
+    Settings settings = {};
+    std::string available_bandwidth;
+    std::string device_token;
+    std::string events_url = production_events_url();
+    std::string language;
+    std::string network_type;
+    std::string geoip_country_path;
+    std::string geoip_asn_path;
+    std::string platform;
+    std::string probe_asn;
+    std::string probe_cc;
+    std::string probe_family;
+    std::string registry_url = production_registry_url();
+    std::string secrets_path = "orchestrator_secrets.json";
+    std::string software_name = "measurement_kit";
+    std::string software_version = MK_VERSION;
+    std::vector<std::string> supported_tests;
+
+    void register_probe(Callback<Error &&> &&callback) const;
+    void update(Callback<Error &&> &&callback) const;
+    void list_tasks(Callback<Error &&, std::vector<Task> &&> &&callback) const;
+};
+
+class Task {
+  public:
+    std::string events_url = production_events_url();
+    std::string task_id;
+
+    void get(Callback<Error &&, std::string &&> &&callback) const;
+    void accept(Callback<Error &&> &&callback) const;
+    void reject(Callback<Error &&> &&callback) const;
+    void done(Callback<Error &&> &&callback) const;
+};
+
+} // namespace orchestrate
+} // namespace ooni
+} // namespace mk
+```
+
+# STABILITY
+
+1 - Experimental
+
+# DESCRIPTION
+
+The orchestrator client communicates with OONI's orchestrator. That is a set
+of services telling clients which tests it is most optimal to run, and with
+what inputs. This allows OONI to maximize censorship coverage given the probe's
+current geographic location and internet service provider.
+
+The `production_registry_url`, `testing_registry_url`, `production_events_url`,
+and `testing_events_url` functions return, respectively, the production and
+testing URLs of the orchestrator's registry and events web services used by
+OONI.
+
+Once you have constructed an orchestration client, you can configure it by
+setting the proper configuration attributes, described below:
+
+The `logger` attribute can be override to use a custom logger. By default, the
+global MeasurementKit logger is used.
+
+The `settings` attribute allows to specify optional settings. By default, an
+empty settings object is used, meaning that defaults will always be used.
+
+The `available_bandwidth` attribute indicates how much bandwidth you would
+like to use for automatic tests run through orchestration. By default, this
+attribute is empty, meaning that no bandwidth quota will be enforced.
+
+The `device_token` attribute uniquely identifies the app inside a device, and
+is only meaningful in the mobile context. This token is generated by the mobile
+OS for the current application and is used to identify it.
+
+The `events_url` attribute contains the URL used to query the `events` web
```

Comment:
Maybe you can link to: https://github.com/TheTorProject/proteus/tree/master/proteus-events